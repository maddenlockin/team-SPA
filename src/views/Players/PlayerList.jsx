import { useEffect, useState } from 'react'
import { Link, useParams } from 'react-router-dom';
import { getPlayers } from '../../services/players';

export default function PlayerList() {
    const { id } = useParams();
    console.log(id);
    const [players, setPlayers] = useState([]);

    useEffect(() => {
        getPlayers()
        .then((response) => setPlayers(response))
    }, []);

    if(!players) return <h1> Loading...</h1>

    return (
        <>
            <ul>
                {players.map((player) => {
                    return <li key={player.id}>
                        <Link to={`/players/${player.id}`} 
                        className= 'App-link'>
                            {player.name}
                        </Link>
                    </li>
                })}
            </ul>
        </>
    )
}
