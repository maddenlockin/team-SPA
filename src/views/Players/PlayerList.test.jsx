import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router"
import PlayerList from "./PlayerLIst";

it('should render a list of players', async () => {
    render(
        <MemoryRouter>
            <PlayerList />
        </MemoryRouter>
    );

    const player1 = await screen.findByText('Kary Chair');
    const player2 = await screen.findByText('Lampy Vampy');


    expect(player1).toBeInTheDocument();
    expect(player2).toBeInTheDocument();
})