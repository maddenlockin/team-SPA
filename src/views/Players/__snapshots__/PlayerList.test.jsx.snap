// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of players 1`] = `
<div>
  <ul>
    <li>
      <a
        class="App-link"
        href="/players/1"
      >
        Boppy Boop
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/players/2"
      >
        Doris The Cat
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/players/3"
      >
        Finney The Elder Cat
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/players/5"
      >
        Kary Chair
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/players/4"
      >
        Lampy Vampy
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/players/6"
      >
        Penny Hen
      </a>
    </li>
  </ul>
</div>
`;
